sources/sqlserver/data.go

```diff
@@ -180,84 +171,28 @@ func convNumeric(conv *internal.Conv, val string) (interface{}, error) {
 // convTimestamp maps a source DB timestamp into a go Time Spanner timestamp
 // It handles both datetime and timestamp conversions.
 func convTimestamp(srcTypeName string, TimezoneOffset string, val string) (t time.Time, err error) {
-	// mysqldump outputs timestamps as ISO 8601, except
-	// it uses space instead of T.
-	if srcTypeName == "timestamp" {
-		// We consider timezone for timestamp datatype.
-		// If timezone is not specified in mysqldump, we consider UTC time.
-		if TimezoneOffset == "" {
-			TimezoneOffset = "+00:00"
+	if srcTypeName == "datetimeoffset" {
+		// val will be in the format "2021-12-15 07:39:52.9433333 +0000 +0000"
+		// the part after time can be ignored
+		if idx := strings.Index(val, "+"); idx != -1 {
+			val = val[:idx-1]
 		}
-		// convert timestamp from format "2006-01-02 15:04:05" to
-		// "2006-01-02T15:04:05+00:00".
 		timeNew := strings.Split(val, " ")
 		timeJoined := strings.Join(timeNew, "T")
 		timeJoined = timeJoined + TimezoneOffset
 		t, err = time.Parse(time.RFC3339, timeJoined)
 	} else {
 		// datetime: data should just consist of date and time.
-		// timestamp conversion should ignore timezone. We mimic this using Parse
-		// i.e. treat it as UTC, so it will be stored 'as-is' in Spanner.
+		// timestamp conversion should ignore timezone.
+		// val will be in this format "2021-12-15 07:39:52.943 +0000 UTC"
+		// the part after time can be ignored
+		if idx := strings.Index(val, "+"); idx != -1 {
```

Comment:
nit: Since this part is common in to both if and else, move this to before the if-else block.
Also, comments convention is that sentences should end with a full stop and start with a capital letter (unless specifying variable/function names that start with a small letter like val in this case).

Reply:
The method that had that block has been updated. Now the query returns the dates in the desired format. So few of the codes were removed.